src/qibo/models/iqae.py

```diff
@@ -0,0 +1,385 @@
+                N_total_plus, N_total_shots, alpha=(alpha / T)
+            )
+            return np.abs(np.arccos(1 - 2 * a_max) - np.arccos(1 - 2 * a_min)) / 2
+
+        y = [
+            (lambda t: h_calc_CP(int(t * N_shots), N_shots, alpha, T))(t)
+            for t in x[tuple(x_domain)]
+        ]
+        L_max = np.max(y) / (2 * np.pi)
+        L_min = np.min(y) / (2 * np.pi)
+        return L_max, L_min
+
+    def calc_L_range_CH(self, N_shots, T):
+        """
+        Calculate the confidence interval for the Chernoff method
+        Args:
+            N_shots: number of shots
+            T: maximum number of rounds to achieve the desired absolute error
+        Returns:
+            L_max, L_min: the maximum and minimum possible error which could be returned
+            on a given iteration
+        """
+        L_max = (
+            np.arcsin((2 / (N_shots) * np.log(2 * T / self.alpha)) ** (1 / 4))
+            / 2
+            / np.pi
+        )
+        L_min = np.arcsin(np.sin(L_max) ** 2)
+        return L_max, L_min
+
+    def find_next_k(self, K_i, up_i, theta_l, theta_u, r=2):
+        """Find the largest integer K such that the interval K*[theta_l,theta_u]
+        lies completely in [0, pi] or [pi, 2pi].
+
+        Args:
+            k: the current power of the Q operator
+            up_i: boolean flag of whether theta_interval lies in the
+                upper half-circle [0, pi] or in the lower one [pi, 2pi]
+            theta_l: the current lower limit of the confidence interval for the angle theta
+            theta_u: the current upper limit of the confidence interval for the angle theta
+
+        Returns:
+            The next power k, and boolean flag for the extrapolated interval
+        """
+        K_max = int(1 / (2 * (theta_u - theta_l)))
+        K = K_max - (K_max - 2) % 4
+        while K >= r * K_i:
+            theta_min = K * theta_l - int(K * theta_l)
+            theta_max = K * theta_u - int(K * theta_u)
+            if (
+                theta_max <= 1 / 2
+                and theta_min <= 1 / 2
+                and int(K * theta_u) == int(K * theta_l)
+            ):
+                up = True
+                return (K, up)
+            elif (
+                theta_max >= 1 / 2
+                and theta_min >= 1 / 2
+                and int(K * theta_u) == int(K * theta_l)
+            ):
+                up = False
+                return (K, up)
+            K = K - 4
+        return (K_i, up_i)
+
+    # EXECUTING THE ALGORITHM
+    def execute(self, backend=None):
+        """Execute IQAE algorithm.
+
+        Args:
+            backend: the qibo backend.
+
+        Returns:
+            An IterativeAmplitudeEstimation results object.
+        """
+        if backend is None:
+            from qibo.backends import GlobalBackend
+
+            backend = GlobalBackend()
+        # Initializing all parameters
+        K = [2]
+        k = [0]
```

Comment:
You have a point there. But I was following the notation from the original paper where they call the variables in that way. So for an experienced reader, it will be easier to establish a relationship between the paper and the code. Do you think `k` and `capital_k` would be better, for example?